assets/helpers/tracking/googleTagManager.js

```diff
@@ -90,28 +103,41 @@ function getContributionValue() {
   return storage.getSession('contributionValue') || 0;
 }
 
+function sendData(event: EventType, participations: Participations, paymentRequestApiStatus: PaymentRequestAPIStatus) {
+  window.googleTagManagerDataLayer.push({
+    event,
+    // orderId anonymously identifies this user in this session.
+    // We need this to prevent page refreshes on conversion pages being
+    // treated as new conversions
+    orderId: getDataValue('orderId', uuidv4),
+    currency: getDataValue('currency', getCurrency),
+    value: getContributionValue(),
+    paymentMethod: storage.getSession('paymentMethod') || undefined,
+    campaignCodeBusinessUnit: getQueryParameter('CMP_BUNIT') || undefined,
+    campaignCodeTeam: getQueryParameter('CMP_TU') || undefined,
+    experience: getVariantsAsString(participations),
+    ophanBrowserID: getOphanIds().browserId,
+    paymentRequestApiStatus,
+  });
+}
+
+
 function pushToDataLayer(event: EventType, participations: Participations) {
   window.googleTagManagerDataLayer = window.googleTagManagerDataLayer || [];
 
-
-  getPaymentAPIStatus()
-    .then((paymentRequestApiStatus) => {
-      window.googleTagManagerDataLayer.push({
-        event,
-        // orderId anonymously identifies this user in this session.
-        // We need this to prevent page refreshes on conversion pages being
-        // treated as new conversions
-        orderId: getDataValue('orderId', uuidv4),
-        currency: getDataValue('currency', getCurrency),
-        value: getContributionValue(),
-        paymentMethod: storage.getSession('paymentMethod') || undefined,
-        campaignCodeBusinessUnit: getQueryParameter('CMP_BUNIT') || undefined,
-        campaignCodeTeam: getQueryParameter('CMP_TU') || undefined,
-        experience: getVariantsAsString(participations),
-        ophanBrowserID: getOphanIds().browserId,
-        paymentRequestApiStatus,
+  try {
+    getPaymentAPIStatus()
+      .then((paymentRequestApiStatus) => {
+        sendData(event, participations, paymentRequestApiStatus);
+      })
+      .catch((e) => {
+        logException(e);
+        sendData(event, participations, 'PromiseRejected');
```

Comment:
Do you need this outer `catch` now that you have the inner one?

Reply:
Maybe not, but since it is a critical flow, I would l prefer to keep it.